jerry-core/vm/vm.c

```diff
@@ -1764,7 +1764,7 @@ vm_loop (vm_frame_ctx_t *frame_ctx_p) /**< frame context */
                                && -(ECMA_INTEGER_MULTIPLY_MAX * ECMA_INTEGER_MULTIPLY_MAX) >= ECMA_INTEGER_NUMBER_MIN,
                                square_of_integer_multiply_max_must_fit_into_integer_value_range);
 
-          if (ecma_are_values_integer_numbers (left_value, right_value))
+          if (left_value != 0 && right_value != 0 && ecma_are_values_integer_numbers (left_value, right_value))
           {
```

Comment:
Ok, it is a bug with the multiplication but we need to move this check down to the `if` starting with:

    if (-ECMA_INTEGER_MULTIPLY_MAX <= left_integer

Add:

    && left_integer != 0
    && right_integer != 0